### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by AbeeraTahir**
In this PR regarding milestone 3 of the OOP School library project, I have:

- Created a class `Classroom` with the following:
  - `@label` instance variable, with its initialization in the constructor.
  - Setter and getter for `@label`.
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`, with the following implementations:
  - Created the `has-many` side (a classroom has many students).
  - Created the `belongs-to` side (a student belongs to a classroom).
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables with their initialization in the constructor.
  - Setters and getters for instance variables.
- Created a class `Rental` with the following:
  - `@date` instance variable, with its initialization in the constructor.
  - Setter and getter for `@date`.
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`, with the following implementation:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require_relative './rental'
+
+class Book
+  attr_reader :rentals
```
Comment (Ismailco) [inline book.rb]: - [x] Getters and Setters for `@rentals` should be present in the Book class, good job adding getters (`attr_reader`) to the `rentals` however you should set up setters also. you can do that by adding the `:rentals` to attr_accessor`

File: person.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 require_relative './nameable'
 require_relative './capitalize_decorator'
 require_relative './trimmer_decorator'
+require_relative './rental'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Ismailco) [inline person.rb]: - [x] Same here Getters and Setters for `@rentals` should be present in the Person class, you should set up also  setters for `:rentals` by adding the `:rentals` to `attr_accessor`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AbeeraTahir, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @AbeeraTahir , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## Highlight
:+1: Readme file
:+1: no linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**